nifi-nar-bundles/nifi-graph-bundle/nifi-neo4j-cypher-service/src/main/java/org/apache/nifi/graph/Neo4JCypherClientService.java

```diff
@@ -0,0 +1,311 @@
+            .displayName("Load Balancing Strategy")
+            .description("Load Balancing Strategy (Round Robin or Least Connected)")
+            .required(false)
+            .defaultValue(LOAD_BALANCING_STRATEGY_ROUND_ROBIN.getValue())
+            .allowableValues(LOAD_BALANCING_STRATEGY_ROUND_ROBIN, LOAD_BALANCING_STRATEGY_LEAST_CONNECTED)
+            .build();
+
+    public static final PropertyDescriptor CONNECTION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-time-out")
+            .displayName("Neo4J Max Connection Time Out (seconds)")
+            .description("The maximum time for establishing connection to the Neo4j")
+            .defaultValue("5 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_POOL_SIZE = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-pool-size")
+            .displayName("Neo4J Max Connection Pool Size")
+            .description("The maximum connection pool size for Neo4j.")
+            .defaultValue("100")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_ACQUISITION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-acquisition-timeout")
+            .displayName("Neo4J Max Connection Acquisition Timeout")
+            .description("The maximum connection acquisition timeout.")
+            .defaultValue("60 second")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor IDLE_TIME_BEFORE_CONNECTION_TEST = new PropertyDescriptor.Builder()
+            .name("neo4j-idle-time-before-test")
+            .displayName("Neo4J Idle Time Before Connection Test")
+            .description("The idle time before connection test.")
+            .defaultValue("60 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_LIFETIME = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-lifetime")
+            .displayName("Neo4J Max Connection Lifetime")
+            .description("The maximum connection lifetime")
```

Comment:
Is there any more documentation on this (and other timeout) property? When would I want to set a connection lifetime at all if it were actively being used? Are the default settings the Neo4J recommended/implemented defaults?

Reply:
I think they are. I think in the previous PR, @mans2singh said that he set them to be roughly the defaults or recommended values.

Reply:
@mattyb149 @MikeThomsen - 
Here is the reference for driver properties including max connection life time (section 2.4.3) [driver config](https://neo4j.com/docs/driver-manual/current/client-applications/#driver-configuration).
Thanks